src/sentry/incidents/grouptype.py

```diff
@@ -99,12 +99,14 @@ def create_occurrence(
     def extract_dedupe_value(self, data_packet: DataPacket[QuerySubscriptionUpdate]) -> int:
         return int(data_packet.packet.get("timestamp", datetime.now(UTC)).timestamp())
 
-    def extract_value(self, data_packet: DataPacket[QuerySubscriptionUpdate]) -> int:
+    def extract_value(
+        self, data_packet: DataPacket[QuerySubscriptionUpdate | AnomalyDetectionUpdate]
+    ) -> int | dict:
```

Comment:
just a heads up - you might need to update the `MetricIssueDetectorHandler` types with this change as well; https://github.com/getsentry/sentry/pull/95613/files#diff-81f569e5989a9b26aad32b54a29a451eb9aa1fb290ffa3b753375db4a8887d0eR39 -- the `int` is passed as a generic and then we'll use it to enforce typing inside of workflow engine.